new_game.py

```diff
@@ -0,0 +1,139 @@
+        answer = self.words[answer_idx]
+        del self.words[answer_idx]
+        return answer
+        
+    # Queries the definition of the answer
+    def definition(self, answer):
+        query = self.lookup.meaning(answer)
+        print '\nDefinition: \n'
+        for definition in query:
+            print definition, '\n', query[definition]
+        print '-' * 75 + '\n'
+    # Builds a list consisting of the answer and 3 other random words
+    def choices(self,answer):
+        my_choices = [answer]
+        while len(my_choices) < 4:
+            choice = random.choice(self.words)
+            if choice not in my_choices:
+                my_choices.append(choice)
+        random.shuffle(my_choices)
+        print my_choices
+    # Prompts user to type the answer 3x if the guess is incorrect
+    def practice(self, answer):
+        print 'Please type the answer 3x, each on its own line.\n'
+        count = 0
+        while count < 3:
+            word = raw_input('> ')
+            if word == answer:
+                count += 1
+            else:
+                print 'Make sure your spelling is correct.'
+        game_on = False
+
+
+class Main(object):
+
+    def __init__(self):
+        self.wins = 0
+        self.attempts = 0
+
+    def clear_screen(self):
+        if os.name == "posix":
+            # Unix/Linux/MacOS/BSD/etc
+            os.system('clear')
+        elif os.name in ("nt", "dos", "ce"):
+            # DOS/Windows
+            os.system('CLS')
+        else:
+            # Fallback for other operating systems.
+            print '\n' * numlines
+    # Prompt to choose easy = number version or hard = word version
+    def start(self):
+        self.clear_screen()
+        game_on = True
+        while game_on:
+            choice = raw_input('\nWould you like the easy or hard version? \n').lower()
+            if choice.startswith('h'):
+                self.reset_score()
+                self.word_game()
+            elif choice.startswith('e'):
+                self.reset_score()
+                self.num_game()
+                print 'Given the definition, choose the correct number that',
+                'corresponds to the word. You have two tries.'
+            else:
+                print 'I don\'t know what that means.'
+    # Choose the number of questions
+    def num_questions(self):
+        while True:
+            try:
+                num = int(raw_input('How many questions would you like? (1-20) '))
+            except:
+                print "That is not a valid number."
+                continue
+            else:
+                break
+        return num
+    # Word game layout
+    def word_game(self):
+        num = self.num_questions()
+        print '\nGiven the definition, type the correct word. You have one try.'
+        raw_input('Press Enter to start...')
+        for i in range(num):
+            time.sleep(2)
+            self.clear_screen()
+            answer = data.answer()
```

Comment:
How does the class know what data is? Nothing inside a class can access
objects outside of it

On Sunday, April 3, 2016, ricebullet <notifications@github.com> wrote:

> In new_game.py
> <https://github.com/katheern/gre_game/pull/1#discussion_r58323488>:
>
> > +                num = int(raw_input('How many questions would you like? (1-20) '))
> > +            except:
> > +                print "That is not a valid number."
> > +                continue
> > +            else:
> > +                break
> > +        return num
> > +    # Word game layout
> > +    def word_game(self):
> > +        num = self.num_questions()
> > +        print '\nGiven the definition, type the correct word. You have one try.'
> > +        raw_input('Press Enter to start...')
> > +        for i in range(num):
> > +            time.sleep(2)
> > +            self.clear_screen()
> > +            answer = data.answer()
>
> I did it outside of any class, at the very bottom, but I'm guessing since
> you asked, I should do it once the game_type is chosen?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/katheern/gre_game/pull/1/files/c8942451c88043a65038c57010977a82b8a29104#r58323488>
>


-- 
Sent from my iPhone